plugins/woocommerce-blocks/assets/js/blocks/product-filters/inner-blocks/attribute-filter/edit.tsx

```diff
@@ -144,7 +149,31 @@ const Edit = ( props: EditProps ) => {
 			</Wrapper>
 		);
 
-	if ( attributeOptions.length === 0 )
+	if ( isLoading )
+		return (
+			<Wrapper>
+				<ul className="is-loading wp-block-woocommerce-product-filter-attribute__loading">
+					{ [ ...Array( 5 ) ].map( ( x, i ) => {
+						return (
+							<li
+								key={ i }
+								style={ {
+									/* stylelint-disable */
+									width:
+										Math.floor(
+											Math.random() * ( 100 - 25 )
+										) + '%',
+								} }
+							>
+								&nbsp;
+							</li>
+						);
+					} ) }
+				</ul>
```

Comment:
@roykho I'm sorry, not sure why this wasn't submitted.

Without using `useMemo`, the bars' width is changed between rerenders.

https://github.com/user-attachments/assets/6f15a9f6-5290-4fc8-91a4-61084f459b6e